onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h

```diff
@@ -54,30 +55,48 @@ Status SkipLayerNormSmallOp(const SkipLayerNormParams<T>* params) {
   return HIP_CALL(hipGetLastError());
 }
 
-#define ADD_OP(threads_per_block)                                         \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 1>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 2>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 4>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 8>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 16>);
+template <typename T, int ThreadsPerBlock, int VecSize>
+Status SkipLayerNormRegularOp(const SkipLayerNormParams<T>* params) {
+  TUNABLE_OP_RETURN_UNSUPPOTED_ARGUMENT_IF(
+      !((params->ld > 0 && params->ld % VecSize == 0 && params->ld >= ThreadsPerBlock * VecSize)));
```

Comment:
I think we need to at least have one kernel registered, that works with any ld size, say 177 (a non-power-of-2 size 177), otherwise, TunableOp won't have any kernel supported.

Reply:
Actually, never mind, "params->ld >= ThreadsPerBlock * VecSize" seems ok.

Reply:
The problem is we not deal with input_size < 64, will add a logical to make sure skiplayernormRegularOp TPB=64 Vec=1 conver all small case.